skcosmo/sample_selection/_base.py

```diff
@@ -346,3 +397,226 @@ def __init__(
+        self.low_dim_idx = low_dim_idx
+
+        if low_dim_idx is None:
+            self.low_dim_idx = [0]
+        else:
+            self.low_dim_idx = low_dim_idx
+
+    def fit(self, X, y):
+        """
+        Learn the samples that form the convex hull.
+
+        Parameters
+        ----------
+        X        : ndarray of shape (n_samples, n_features)
+                   Feature matrix of samples to use for constructing the convex
+                   hull.
+        y        : ndarray of shape (n_samples,)
+                   Target values (property on which the convex hull should be
+                   constructed, e.g. Gibbs free energy)
+
+        Returns
+        -------
+        self : object
+            Fitted scorer.
+        """
+
+        X, y = check_X_y(X, y, ensure_min_features=2)
+        self.n_features_in_ = X.shape[1]
+
+        if len(y.shape) == 1:
+            y = y.reshape((len(y), 1))
+
+        if (max(np.abs(self.low_dim_idx)) > X.shape[1]) and (
+            min(self.low_dim_idx) >= 0
+        ):
+            raise ValueError(
+                "One or more columns indexed with low_dim_idx is"
+                " out of bounds with the dimensions of X."
+            )
+
+        self.high_dim_idx_ = np.setdiff1d(np.arange(X.shape[1]), self.low_dim_idx)
+
+        # get number of dimensions for the convex (lower dimensional) hull construction
+        n_low_dim_idx = len(self.low_dim_idx)
+
+        # append features and target property to the same data matrix (for the
+        # convex hull construction)
+        convex_hull_data = np.zeros((X.shape[0], n_low_dim_idx + 1))
+        convex_hull_data[:, :1] = y
+        convex_hull_data[:, 1:] = X[:, self.low_dim_idx].copy()
+
+        # create high-dimensional feature matrix
+        high_dim_feats = X[:, self.high_dim_idx_]
+        # get scipy convex hull
+        convex_hull = ConvexHull(convex_hull_data)
+        # get normal equations to the hull simplices
+        y_normal = convex_hull.equations[:, 0]
+
+        # get vertices_idx of the convex hull
+        vertices_idx = np.unique(
+            convex_hull.simplices[np.where(y_normal < 0)[0]].flatten()
+        )
+
+        # assign vertices_idx as internal param
+        self.vertices_idx_ = vertices_idx
+
+        # required for the score_feature_matrix function
+        self.interpolator_high_dim_ = _linear_interpolator(
+            points=convex_hull_data[self.vertices_idx_, 1:],
+            values=high_dim_feats[self.vertices_idx_],
+        )
+
+        # required to compute the distance of the low-dimensional feature to the convex hull
+        self.interpolator_y_ = _linear_interpolator(
+            points=convex_hull_data[self.vertices_idx_, 1:],
+            values=convex_hull_data[self.vertices_idx_, 0],
+        )
+
+        return self
+
+    def _check_is_fitted(self, X):
+        check_is_fitted(
+            self, ["high_dim_idx_", "interpolator_high_dim_", "interpolator_y_"]
```

Comment:
should also have `vertices_idx_` or equivalent

Reply:
added